pkg/cache/memory/memory.go

```diff
@@ -0,0 +1,77 @@
+package memory
+
+import (
+	"strings"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+)
+
+const (
+	expirationInterval = 12 * time.Hour
+	purgeInterval      = 13 * time.Hour
+	defaultExpiration  = cache.DefaultExpiration
+)
+
+// Cache is a wrapper around the go-cache library.
+type Cache struct {
+	c *cache.Cache
+}
+
+// New constructs a new in-memory cache.
+func New() *Cache {
+	c := cache.New(expirationInterval, purgeInterval)
+	return &Cache{c: c}
+}
+
+// NewWithData constructs a new in-memory cache with existing data.
+func NewWithData(ctx context.Context, s string) *Cache {
+	data := strings.Split(s, ",")
```

Comment:
**Suggestion:** This interface is a bit too lossy; I think it would be easier to reason about if `NewWithData` took a varargs for the items instead of an encoded string. As an example (and we may never run into this but..) how would one initialize a cache with a comma in the key?